### PR TITLE
improve IntegralGrader's handling of complex numbers

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
This PR improves IntegralGrader's handling of complex numbers. The integrator we use (scipy.integrate.quad) cannot handle complex limits. So:

- Raise a ConfigError if author-specified upper/lower limits evaluate to complex
- Raise an IntegrationError if learner-specified upper/lower limits evaluate to complex

The `scipy.integrate.quad` integrator also cannot handle complex integrands by itself. But we can pass real and imaginary parts to the integrator separately. So this PR adds a configuration option `complex_integrand` (default is False) such that:

- if `complex_integrand:False` and integrand evaluates to complex, raise IntegrationError
- if `complex_integrand:True`, evaluate real/imaginary parts separately and compare author/learner values.

Debug messages and tests also updated.

### CODE REVIEW

**Comment by ChristopherChudzicki** [comment]
@jolyonb This is ready for review, although I have not updated version number, python_lib, or change-log. (Note: #27 also did not update these things ... )

**Comment by ChristopherChudzicki** [comment]
@jolyonb Oh, I forgot to update the documentation. Maybe don't review yet.

**Comment by ChristopherChudzicki** [comment]
@jolyonb Apparently IntegralGrader's only documentation is its docstring, which I have now updated. So ready for review.

File: mitxgraders/plugins/integralgrader.py [hunk 1]
```diff
@@ -338,42 +366,67 @@ def raw_check(self, answer, cleaned_input):
...
+                author_re_error=expected_re[1],
+                author_re_neval=expected_re[2]['neval'],
+                author_im_eval=expected_im[0],
+                author_im_error=expected_im[1],
+                author_im_neval=expected_im[2]['neval'],
             ))
 
             # Check if expressions agree
-            if not within_tolerance(expected[0], student[0], self.config['tolerance']):
+            if not within_tolerance(expected_re[0], student_re[0], self.config['tolerance']):
```
Comment (jolyonb) [inline mitxgraders/plugins/integralgrader.py]: Note that within_tolerance can handle complex numbers. Perhaps make a check value for a single within_tolerance call, so you don't have to repeat the consequences?

File: mitxgraders/plugins/integralgrader.py [hunk 2]
```diff
@@ -69,12 +70,22 @@ class IntegrationError(Exception):
     """Represents an error associated with integration"""
     pass
 
+def check_output_is_real(func, error, message):
```
Comment (jolyonb) [inline mitxgraders/plugins/integralgrader.py]: Please use wraps when making a decorator! https://www.blog.pythonlibrary.org/2016/02/17/python-functools-wraps/

**Comment by jolyonb** [comment]
Sorry this took so long. Two small comments, but otherwise, looks good!

**Comment by ChristopherChudzicki** [comment]
@jolyonb Sorry *this* took so long :). 

I made the changes you suggested and added some tests. Looking at the test file, I realized that all the tests I wrote were for complex limits or errors related to complex integrands. I forgot to add (until now) tests that assert complex_integrands behave correctly when expected.

Thanks for `@wraps` tip, that's cool. There are probably some other places in the codebase that could benefit from `@wraps`.

**Comment by ChristopherChudzicki** [comment]
@jolyonb If you have a moment, I think this is ready for merge.

**Comment by jolyonb** [comment]
Sorry, my bad. I've been on vacation in Australia for two weeks ;-) I'll try to get onto it this evening.

**Comment by jolyonb** [comment]
Sorry about the delay here!

**Pull request merged**